misc.go

```diff
@@ -14,11 +14,47 @@
 package parser
 
 import (
+	"fmt"
+	"strconv"
 	"strings"
 
 	"github.com/pingcap/parser/charset"
 )
 
+// CommentCodeVersion is used to track the highest version can be parsed in the comment with pattern /*T!00001 xxx */
+type CommentCodeVersion int
+
+const (
+	CommentCodeNoVersion  CommentCodeVersion = iota
+	CommentCodeAutoRandom CommentCodeVersion = 40000
+
+	CommentCodeCurrentVersion
+)
+
+func (ccv CommentCodeVersion) String() string {
+	return fmt.Sprintf("%05d", ccv)
+}
+
+func extractVersionCodeInComment(comment string) CommentCodeVersion {
+	if SpecVersionCodePattern.MatchString(comment) {
+		verDigitNum := 5
+		if isDigit(rune(comment[len("/*T!xxxxx")])) {
+			verDigitNum += 1
+		}
```

Comment:
Please use regex captures ([`(*Regexp).FindStringSubmatch`](https://godoc.org/regexp#Regexp.FindStringSubmatch)) to extract the digits instead of this...

Reply:
Updated. PTAL